presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -254,14 +255,18 @@ public PinotQueryGeneratorContext withTopN(LinkedHashMap<VariableReferenceExpres
             checkSupported(!hasAggregation(), "Pinot doesn't support ordering on top of the aggregated data");
         }
         int intLimit = checkForValidLimit(limit);
+
+        LinkedHashMap<VariableReferenceExpression, OrderingColumnInformation> orderByColumnInformation = new LinkedHashMap<>();
```

Comment:
LinkedHashMap is an implementation, use Map in declaration:
Map<VariableReferenceExpression, OrderingColumnInformation> orderByColumnInformation = new LinkedHashMap<>();

Reply:
I disagree with this :-) I prefer LinkedHashMap as a static type because it conveys the order sensitiveness. It signifies that order is important and that is indeed the approach we have taken with other field variables of this class.

Reply:
could be a separate discussion: just see a number of LinkedHashMap in this class for declaration.
are we using LinkedHashMap's order sensitiveness anywhere? I think make LinkedHashMap as an implementation is a strong indication that we would like to keep the order sensitiveness. what do you think?

Reply:
I think it is not a strong guarantee of type safety: The caller can easily pass in a simple Map to the underlying type instead of a LinkedHashMap and that could mess up the order.

I wish Java had some sort of an interface for iteration sensitive Maps, since it doesn't I am advocating for letting the static type of the field remain LinkedHashMap to avoid stamping into it a non LinkedHashMap that messes up the order. 

You would agree that such sort of bugs can be hard to track down and that having strong static type safety here would prevent them.

Reply:
could you please add more details, or some example code to show why order is important and needs to be kept here? I did not find LinkedHashMap's order sensitiveness is used in this class

Reply:
I think the order by columns stored here in `orderByColumnInformation` itself is a valid example as we would need to sort in the order we are seeing the columns. i.e. for  `order by secondsSinceEpoch desc, 1 desc` we need to preserve order of secondsSinceEpoch and `city` to satisfy the comparison requirement in sorting

Reply:
get it. Do you think SortedMap and TreeMap could be a candidate to keep the order? We could do it in followup PR.